src/Iris/Cli/Interactive.hs

```diff
@@ -46,3 +50,21 @@ interactiveModeP = nonInteractiveP <|> pure Interactive
         [ Opt.long "no-input"
         , Opt.help "Enter the terminal in non-interactive mode"
         ]
+
+{- | Forces non interactive mode when the terminal is not interactive
+
+Use this function to check whether you can get input from the terminal:
+
+@
+'handleInteractiveMode' 'requestedInteractiveMode'
+@
+
+If the terminal is non interactive i.e. the program is run in a pipe, interactive mode is set to false no matter what
+
+@since 0.0.0.1
+-}
+handleInteractiveMode :: InteractiveMode -> IO InteractiveMode
+handleInteractiveMode optionMode = do
+    isTerminal <- hIsTerminalDevice stdin -- misleading on windows where some terminals are emulated
+    supportsANSI <-  hSupportsANSI stdin -- false on dumb terminal, better windows support
```

Comment:
It's better to check `stderr`. This check will fail on redirecting stdout to some other tool.

Ideally, we want separate interactive flags for `stdout` and `stderr` but this might be too complicated for now 🤔

Reply:
inputs come from stdin only
If I redirect stdout and stderr I can still respond to questions, without seeing them
Or even seeing them if I do something like:
```bash
mycli 2>&1 | tee clilog
```
Anyway my main use case for interactivity is:

* on interactive terminals ask questions
* on non interactive ones pick a default behaviour

Testing both stderr and stdin would indeed make sense

Reply:
as by the discussion below, and [guidelines](https://clig.dev/#interactivity) I'm now testing only `hSupportsANSI` on stdin
I believe your last sentence on checking stdout is a typo as contraddicts the guidelines